vignettes/design_principle.Rmd

```diff
@@ -0,0 +1,90 @@
+
+
+
+There are several R packages designed to access data from specific HIS. They are usually tied to a single HIS and can't be used to query others. {fingertipsR}, {REDCapR}, {godataR}, {globaldothealth} can be used to fetch data from [Fingertips](https://fingertips.phe.org.uk/), [REDCap](https://projectredcap.org/software/), [goData](https://www.who.int/tools/godata), and [Global.Health](https://global.health/) respectively. In these packages, the API end points are accessed differently. This introduces the challenge of having a unified framework for importing data from these HIS into R.
+
+To address this challenge, PIs of the [Epiverse-TRACE](https://data.org/initiatives/epiverse/) came with a list of potential data sources for which we aim at building a tool to make requests and fetch the data of interest in the same way. The data sources include: distributed health information systems, and public databases as shown in the figure below.
+
+![readepi roadmap](../man/figures/roadmap-readepi.drawio.png)
+
+## Scope
+
+The {readepi} package intends to import data from two common sources of institutional health-related data: health information systems (HIS) wrapped with specific program interfaces (APIs) and relational database management systems (RDBMS) that run on specific servers. Each source has its own import function, `read_from_api()` for HIS and `read_from_server()` for RDBMS. There is also an auxiliary function, `get_metadata()`, that retrieves metadata from a specified API and offers interoperability with the [{cleanepi} R package](https://epiverse-trace.github.io/cleanepi/).     
+The current version of {readepi} supports importing data from HIS APIs such as REDCap (Research Electronic Data Capture), DHIS2 (District Health Information System 2), Fingertips, GoData, and Globaldothealth, as well as RDBMS such as MS SQL, MS ACCESS, SQlite, MYSQL, and PostgreSQL. 
+
+![readepi design](../man/figures/readepi_design.svg)
+
+
+## Output
+
+The two import functions return a `list` object containing one or more `data frames`. Each `data frame` corresponds to the data from a specified table or file. The `get_metadata()` function returns a data dictionary containing information about the data structure. 
+
+## Design decisions
+
+The aim of {readepi} is to simplify and standardize the process of fetching health data from APIs and servers. We ambition to make it requires minimal information to access and pull the data of interest from the target source.  
+
+1. The `read_from_api()` function takes only two arguments to import data from an API: an object of class **credential** and an object of class **api**.
```

Comment:
This sounds good.

For now, we will explicitly provide the name of the HIS using the `name` argument. This can be modified later on if we realised that the API names can be identified from the URL.